### PR TITLE
Kennric/pr comment script

### PR DESCRIPTION
**Pull request by Kennric**
This adds a simple script for adding comments to a github PR

### CODE REVIEW

**Comment by Kennric** [comment]
@ramereth

**Comment by ramereth** [comment]
@Kennric can I assume the jenkins job and logic for the pelican site commenting will be in a different PR?

**Comment by Kennric** [comment]
@ramereth yes, this is just the comment script itself. Currently the pelican jobs are manually created, to be replaced with cookbook generated jobs when the script and workflow is verified to work as expected.

File: files/default/bin/github_comment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/opt/chef/embedded/bin/ruby
+require_relative '../lib/github_comment.rb'
+
+# quit unless we get three command line arguments
+unless ARGV.length == 3
+  puts "Requires three arguments.\n"
+  puts "Usage: gh_comments.rb <repo_name> <issue_id> <comment_text>\n"
```
Comment (ramereth) [inline files/default/bin/github_comment.rb]: The name here doesn't match what's used. I.e. it should say ``Usage: github_comment.rb``.

File: test/integration/github_comment/serverspec/github_comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  it { should be_owned_by 'jenkins' }
+  it { should be_grouped_into 'jenkins' }
+end
+
+describe file('/var/lib/jenkins/lib/github_comment.rb') do
+  it { should be_file }
+  it { should be_mode 440 }
+  it { should be_owned_by 'jenkins' }
+  it { should be_grouped_into 'jenkins' }
+end
```
Comment (ramereth) [inline test/integration/github_comment/serverspec/github_comment_spec.rb]: If you rebase to the latest master you should also add the following to ensure the jenkins server is working as it should:

``` ruby
require 'spec_helper'

describe 'github_comment' do
  it_behaves_like 'jenkins_server'
end
```

File: spec/unit/recipes/github_comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          }
+          node.set['osl-jenkins']['credentials']['jenkins'] = {
+            'github_comment' => {
+              user: 'manatee',
+              pass: 'password',
+              trigger_token: 'trigger_token'
+            }
+          }
+        end.converge(described_recipe)
+      end
```
Comment (ramereth) [inline spec/unit/recipes/github_comment_spec.rb]: You need to add ``include_context 'common_stubs'`` so that these tests now run properly.

**Comment by Kennric** [comment]
commits squashed

**Comment by Kennric** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v0.12.0.  Have a nice day!

**Pull request merged**